calcom/cal.com
# Improve booking seats changes

## What does this PR do?



This PR continues #5427 by updating calendar & video events whenever there are changes in seats bookings. This abstracts some of the seats logic to a `handleSeats` function.


Fixes # (issue)



**Environment**: Staging(main branch) / Production

## Type of change





## How should this be tested?



- Create a booking with seats and add a destination calendar
- Fill the booking with attendees
- As an attendee
    - Move to a different time slot that has no booking
        - There should be a new booking created and the destination calendar should be updated with the new calendar event and the previous one should have that attendee removed
    - For that new booking you should be the only attendee. Reschedule to another empty time slot
        - On the calendar, that previous booking should be removed and a new one created
    - Reschedule back to the booking that has already filled with seats
        - The calendar event should now include the new attendee
    - Now cancel the event as the attendee
        - The calendar event should still be there but now without the attendee
- As the owner
    - Reschedule the booking
        - The whole booking should have been moved
    -  Cancel the booking
        - The whole booking should have been canceled

## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
